sdf.md

```diff
@@ -139,7 +139,46 @@ The target namespace is the namespace into which the defined terms are added. Th
 
 For example if the default namespace in the example above is "foo", then you could use "temperatureData" to refer to the property defined at the URI:
 ```
-https://example.com/#temperatureData
+https://example.com/#odmData/temperatureData
+```
+
+## Optionality using the keyword "required"
+
+The keyword "required" is provided to apply a constraint for which definitions are mandatory in an instance conforming to a particular definition in which the constraint appears.
+
+The value of "required" is an array of reference objects, each of which contains a JSON pointer to one required definition.
+
+The example uses [Relative JSON Pointer][] syntax to point to nearby definitions.
+
+``` json
+{
+  "odmObject": {
+    "temperatureWithAlarm": {
```

Comment:
F2F$: All $ref in the "required", "inputData" and "outputData" should be removed.